homeassistant/components/switch/enigma.py

```diff
@@ -0,0 +1,173 @@
+"""
+homeassistant.components.switch.enigma
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Enable or disable standby mode on Enigma2 receivers.
+
+You should have a recent version of OpenWebIf installed.
+
+There is no support for username/password authentication
+at this time.
+
+Configuration:
+
+To use the switch you will need to add something like the
+following to your config/configuration.yaml
+
+switch:
+    platform: enigma
+    name: Vu Duo2
+    host: 192.168.1.26
+    port: 80
+
+Variables:
+
+host
+*Required
+This is the IP address of your Enigma2 box. Example: 192.168.1.32
+
+port
+*Optional
+The port your Enigma2 box uses, defaults to 80. Example: 8080
+
+name
+*Optional
+The name to use when displaying this Enigma2 switch instance.
+
+"""
+from homeassistant.const import ATTR_FRIENDLY_NAME
+from homeassistant.helpers.entity import ToggleEntity
+from homeassistant.const import STATE_ON, STATE_OFF, CONF_HOST
+import logging
+import requests
+from xml.etree import ElementTree
+
+_LOGGING = logging.getLogger(__name__)
+DOMAIN = "enigma"
+
+# pylint: disable=unused-argument
+
+
+def setup_platform(hass, config, add_devices_callback, discovery_info=None):
+    """ Find and return enigma2 boxes. """
+
+    host = config.get(CONF_HOST, None)
+    port = config.get('port', "80")
+    name = config.get('name', "Enigma2")
+
+    if not host:
+        _LOGGING.error('Missing config variable-host')
+        return False
+
+    add_devices_callback([
+        EnigmaSwitch(name, host, port)
+    ])
+
+
+class EnigmaSwitch(ToggleEntity):
```

Comment:
Instead of having this as a switch, wouldn't it make more sense to make this part of the media player component?